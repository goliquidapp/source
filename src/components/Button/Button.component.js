import React, {Component} from 'react';
import {View, TouchableOpacity, Text} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import Theme from '../../resources/Theme.js';

class Button extends Component{
	render(){
		const {onPress, buttonStyle, textStyle, text, children, colors}=this.props;
		if (colors){
			return (
				<LinearGradient colors={colors} style={[styles.button,buttonStyle, styles.gradient]}>
					<TouchableOpacity onPress={onPress} style={styles.touchContainer}>
						<Text style={[styles.text,textStyle]}>{text}</Text>
					</TouchableOpacity>
				</LinearGradient>
			)
		}
		else{
			return (
				<TouchableOpacity onPress={onPress} style={[styles.button,buttonStyle]}>
					<Text style={[styles.text,textStyle]}>{text}</Text>
					{children}
				</TouchableOpacity>
			)
		}
	}
}

const styles={
	button:{
		flexDirection:'row',
		backgroundColor:Theme['dark'].highlighted,
		paddingVertical:10,
		paddingHorizontal:30,
		elevation:5,
		alignItems:'center',
		justifyContent:'center',
		borderRadius:8
	},
	text:{
		color:Theme['dark'].primaryText,
		fontSize:20,
		fontFamily:Theme['dark'].fontNormal
	},
	gradient:{
		elevation:6,
		borderRadius:8
	},
	touchContainer:{
		width:'100%',
		justifyContent:'center',
		alignItems:'center'
	}
}

export default Button;
