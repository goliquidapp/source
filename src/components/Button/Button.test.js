import React from 'react';
import { Text, View, TouchableOpacity} from 'react-native';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme'

import Button from './Button.component';

describe("Testing Button component",()=>{
	it('renders correctly', () => {
		const wrapper = renderer.create(
    		<Button text={"Click me!"}>
                <View>
                    <Text>I am a child</Text>
                </View>
            </Button>
    	).toJSON();
		expect(wrapper).toMatchSnapshot();
	});

    it('call function when clicked',()=>{
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');

        const wrapper = shallow(
            <Button text={"Click me!"} onPress={onPressEvent}>
                <View>
                    <Text>I am a child</Text>
                </View>
            </Button>
        )

        wrapper.find(TouchableOpacity).first().props().onPress();
        expect(onPressEvent.mock.calls.length).toBe(1);
    })
})