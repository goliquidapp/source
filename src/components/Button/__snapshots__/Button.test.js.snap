// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component renders correctly 1`] = `
<View
  accessible={true}
  clickable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5367FF",
      "borderRadius": 8,
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
        },
        undefined,
      ]
    }
  >
    Click me!
  </Text>
  <View>
    <Text>
      I am a child
    </Text>
  </View>
</View>
`;
